### PR TITLE
METRON-990: Clean up and organize flux properties

### PR DESCRIPTION
**Pull request by merrimanr**
## Contributor Comments
This PR is mainly a refactor of the enrichment and indexing flux files along with their matching property files.  The changes include:

- moving important settings with hardcoded values in flux files to property files so that they are configurable through Ambari
- removing old and unused properties
- elasticsearch.properties file is now implemented as a jinja2 template in the mpack, matching the enrichment.properties implementation
- global.json is now implemented as a jinja2 template in the mpack
- properties are now organized in Ambari as separate tabs and sub sections, hopefully making them easier to find
- changed a couple properties to use a dropdown widget in Ambari

I wrote descriptions for new properties in a similar style as existing properties.  I feel the descriptions are a little short, curious if others agree.  I also stopped short of improving ALL our properties with better widgets than just a text box.  I imagine people will have opinions on how to best present properties in Ambari but this is a start.

This has been tested on full dev with the usual process.  When reviewing, spin up full dev and navigate to the Metron service in Ambari.  The Config section should look as described in the section above.  The enrichment.properties and elasticsearch.properties should look much shorter and easier to read.

One more thing.  I added a rat exception for *.json.j2 since we already have an exception for *.json (no comments in json is the reason?).  Let me know if that's wrong.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root incubating-metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by merrimanr** [comment]
Still working on updating the integration tests with the new properties.  Will have them fixed shortly.

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml [hunk 1]
```diff
@@ -253,69 +253,295 @@
...
+        <display-name>Elasticsearch Date Format</display-name>
+    </property>
+    <property>
+        <name>parser_error_topic</name>
+        <description>Parser Error Topic</description>
+        <value>indexing</value>
+        <display-name>Parser Error Topic</display-name>
+    </property>
+    <property>
+        <name>enrichment_workers</name>
```
Comment (nickwallen) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: After deploying in Full Dev, I went into Ambari and changed some Enrichment properties like "Enrichment Workers".  After saving the changes, Ambari indicated that I needed to restart all of the Metron-related components.  Even unrelated components like the Metron Management UI.  It only really needed to restart the Enrichment topology. 

I am not sure what logic controls that.  Is that something controlled in this PR?  Or is that external to these changes?

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml [hunk 2]
```diff
@@ -253,69 +253,295 @@
...
+        <description>Number of Enrichment Topology Workers</description>
+        <value>1</value>
+        <display-name>Enrichment Workers</display-name>
```
Comment (nickwallen) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: It does not seem like changes I make in Ambari are being propagated into the running topologies.  

Here is what I am seeing. After deploying in Full Dev, I went into Ambari and changed some Enrichment properties like "Enrichment Workers".  The Enrichment component was then restarted.  

* In Ambari, the "Enrichment Workers" value is set to 2.  This is what I expect.

* When I look on disk `/usr/metron/0.4.0/config/enrichment.properties` the value is still the default of 1.  I would expect to see 2 here.
        ```
         enrichment.workers=1
        ```

* The Storm UI also indicates that the Enrichment topology is running with a `topology.workers` value of 1.  I would expect to see 2 here.

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml [hunk 3]
```diff
@@ -253,69 +253,295 @@
...
+        <name>bolt_hdfs_rotation_policy_count</name>
+        <description>HDFS Rotation Policy Count</description>
+        <value>1</value>
+        <display-name>HDFS Rotation Policy Count</display-name>
+    </property>
+    <property>
+        <name>bolt_hdfs_rotation_policy_units</name>
+        <description>Frequency of HDFS File Rotation</description>
+        <value>DAYS</value>
+        <display-name>HDFS Rotation Policy Units</display-name>
```
Comment (merrimanr) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: Fixed by adding BYTES, MB, GB, TB to the list of options for this property.  Is that sufficient?

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml [hunk 4]
```diff
@@ -253,69 +253,295 @@
...
+                </entry>
+                <entry>
+                    <value>DAYS</value>
+                </entry>
+            </entries>
+            <selection-cardinality>1</selection-cardinality>
+        </value-attributes>
+    </property>
+    <property>
+        <name>indexing_error_topic</name>
```
Comment (merrimanr) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: Fixed

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/templates/enrichment.properties.j2 [hunk 5]
```diff
@@ -16,72 +16,48 @@
...
-
-##### Metrics #####
-
-#reporters
-org.apache.metron.metrics.reporter.graphite=true
-org.apache.metron.metrics.reporter.console=false
-org.apache.metron.metrics.reporter.jmx=false
-
-#Graphite Addresses
+kafka.security.protocol=PLAINTEXT
```
Comment (merrimanr) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/templates/enrichment.properties.j2]: Fixed

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml (hunk 2 above)
Comment (merrimanr) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: Fixed

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml (hunk 1 above)
Comment (merrimanr) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: This issue existed before this PR but I think it's annoying and would be happy to fix it.  I will have to reorganize metron-env.xml into component-level files so I can make the configuration dependencies more granular.  Will try to get this turned around quickly.

Comment (nickwallen) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: Great.  Feel free to do as a separate PR, if you like.  Definitely not needed as part of this one.

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml (hunk 3 above)
Comment (nickwallen) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: I think the values would be GB, KB, MB, TB  based on what[ I am reading here.](https://storm.apache.org/releases/1.0.1/javadocs/org/apache/storm/hdfs/bolt/rotation/FileSizeRotationPolicy.Units.html)

Comment (merrimanr) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: Sorry you're right.  I actually did use those values.  The comment is wrong.

Comment (nickwallen) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/configuration/metron-env.xml]: [Based on some digging](https://storm.apache.org/releases/1.0.1/javadocs/org/apache/storm/hdfs/bolt/rotation/package-summary.html), there are only 3 policies that can be used; `TimedRotationPolicy`, `FileSizeRotationPolicy`, and `NoRotationPolicy`, and I think you will have covered them all.

**Comment by nickwallen** [comment]
Your updates look good @merrimanr. Just running it up again for a round of testing.

**Comment by nickwallen** [comment]
+1  I was able to alter settings for the Enrichment topology, successfully restart all services, and see the new configuration changes take effect.

Splitting out all of the settings across different tabs is a really nice way to do it.  Would have been a mess if you hadn't gone that extra mile.  Nice

**Pull request closed without merging**